### ISSUE TITLE
Doesn't seem to work with wasm32-unknown-unknown

### ISSUE THREAD

**Issue by d3lm** [wvwwvwwv/scalable-concurrent-containers#116]
I tried to use `scc` for a crate that we compile to `wasm32-unknown-unknown`. I tried to use a `HashIndex` as a static item and while it compiles just fine, it crashes at runtime if I try to insert an item.

Here's what I did:

```rust
static HASH_INDEX: Lazy<HashIndex<u32, u32>> = Lazy::new(|| HashIndex::default());

#[wasm_bindgen)]
pub fn some_fn() -> Result<...> {
  HASH_INDEX.insert(1, 2)?;
}
```

It then crashes at runtime with the following error:

```
panicked at .../.cargo/registry/src/index.crates.io-6f17d22bba15001f/scc-2.0.4/src/hash_table/bucket_array.rs:55:13:
assertion `left == right` failed
  left: 16
 right: 0
```

Here's where it throws:

![CleanShot 2023-12-07 at 17 44 32@2x](https://github.com/wvwwvwwv/scalable-concurrent-containers/assets/12571019/491b4f18-607b-4b82-8668-687f2c0b583e)

Any idea what could be causing this?

**Comment by d3lm** [comment]
Oh yea, that is very good point indeed. The alignment is prolly wrong.

**Comment by wvwwvwwv** [comment]
Hi, thanks for reporting this issue!

The alignment is definitely different from the actual alignment, but that is intended; if the alignment is above a certain threshold (~4?), `alloc_zeroed` becomes awfully slow due to a call to `memset`, otherwise `alloc_zeroed` is simply translated into `calloc` which takes full advantage of demand paging.

I think the problem is related to warm*32* because this crate has never been tested on a 32-bit machine.

I'll try to fix it on this weekend!

**Comment by wvwwvwwv** [comment]
Reproduced the issue and fixed it.

Cause: the size of the `Bucket` struct is no longer a power of two in a 32-bit binary.

I'll cleanup the change, add more tests, and then upload SCC 2.0.5 soon.

**Comment by wvwwvwwv** [comment]
The commit message is wrong, but the issue was fixed in https://github.com/wvwwvwwv/scalable-concurrent-containers/commit/33e4fc5e90ed3313dafeea9c3863724ac57e3ac7

**Comment by wvwwvwwv** [comment]
Fixed in 2.0.5!

-> Theoretically, everything shall work as expected in a 32-bit binary.
-> If you find anything unexpected, please open a new issue, or reopen this one in case it's related to this.

**Comment by d3lm** [comment]
Oh wow, thanks so much for that insanely fast turnaround! I really appreciate it 🙏
